Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -67,22 +67,18 @@ GMallocSpan<T, Malloc> adoptGMallocSpan(std::span<T> span)
     return adoptMallocSpan<T, Malloc>(span);
 }
 
-template<typename Malloc = GMalloc>
-GMallocSpan<char, Malloc> adoptGMallocString(char* str, size_t length)
-{
-    return adoptGMallocSpan<char, Malloc>(unsafeMakeSpan(str, length));
-}
-
-template<typename Malloc = GMalloc>
-GMallocSpan<char, Malloc> adoptGMallocString(char* str)
+template<typename T, typename Malloc = GMalloc>
+GMallocSpan<T, Malloc> dupGMallocSpan(std::span<const T> span)
 {
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-    return adoptGMallocSpan<char, Malloc>(unsafeMakeSpan(str, str ? strlen(str) : 0));
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+    T* newMemory = static_cast<T*>(Malloc::malloc(sizeof(T) * span.size()));
+    auto newSpan = unsafeMakeSpan(newMemory, span.size());
+    memcpySpan(newSpan, span);
+    return adoptMallocSpan<T, Malloc>(newSpan);
```

Comment:
This could use `GMallocSpan::malloc()` and `std::span::size_bytes()`:


```suggestion
    auto duplicate = GMallocSpan<T, Malloc>::malloc(span.size_bytes());
    memcpySpan(duplicate.mutableSpan(), span);
    return duplicate;
```